src/prof.c

```diff
@@ -2691,6 +2691,20 @@ prof_log_emit_metadata(emitter_t *emitter) {
 	emitter_json_object_end(emitter);
 }
 
+/*
+ * The buffer doesn't need a lock, since only one thread is able to use it at a
+ * time due to the protection by prof_logging_state.
+ */
+#define PROF_LOG_STOP_BUFSIZE PROF_DUMP_BUFSIZE
+static char prof_log_stop_buf[
```

Comment:
Since prof_log isn't used by default, even with prof:true, I'd say allocating the buffer in prof_log_start(), e.g. `if (prof_log_stop_buf == NULL) { ...`, so we only pay the price when in use.